Run failed on JDK 8

System with JDK 11 installed and the following Gradle configuration:
```
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(8)
        vendor = JvmVendorSpec.ADOPTOPENJDK
    }
}
```
Got this error:
```
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
Unrecognized option: --add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED
```

Sorry you're having trouble! I'm having difficulty reproducing the issue you're describing. 

I tried adding the code you described to a simple sample Gradle project that uses the plugin; you can find that simple project here: https://github.com/kelloggm/sample-gradle-project/tree/repro-cf-gradle-149. When I run `./gradlew clean build` on the project linked above, I get the output:
```
> Configure project :
Unable to determine Checker Framework version. Assuming default is being used: 3.11.0

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/kelloggm/jsr308/sample-gradle-project/build.gradle' line: 33

* What went wrong:
A problem occurred evaluating root project 'sample-gradle-project'.
> No signature of method: build_22n4gkxop7tl5d7vh5g33jvft.java() is applicable for argument types: (build_22n4gkxop7tl5d7vh5g33jvft$_run_closure3) values: [build_22n4gkxop7tl5d7vh5g33jvft$_run_closure3@62ad181d]
  Possible solutions: wait(), any(), wait(long), tap(groovy.lang.Closure), each(groovy.lang.Closure), any(groovy.lang.Closure)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

```

This output suggests to me that I didn't reproduce the issue you're describing correctly, since the error message indicates that the build file isn't valid. Can you help me produce a valid build file that exhibits this problem, so that I can fix it?

On a theoretical note, I have some suspicions about what might have caused this bug. Without the ability to reproduce it, I can't test them, so what follows is pure speculation. The error message you're seeing definitely suggests a mismatch between the JVM version and the options being passed to the Checker Framework. The option `--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED` is passed by the plugin to the JVM in which the Checker Framework will run if the plugin detects that the JVM is Java9-compatible, according to Gradle's API. It seems to me that you're running a Java 11 JVM in a configuration where it's pretending to be a Java 8 JVM, and thus rejecting the option. One possible solution would be to add an option to the plugin that forces it to assume the underlying JVM is a particular version, rather than relying on the Gradle API; that's not particularly elegant, but I think it would let you work around this particular problem.

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ Add the following to your `build.gradle` file:
 ```groovy
 plugins {
     // Checker Framework pluggable type-checking
-    id 'org.checkerframework' version '0.5.18'
+    id 'org.checkerframework' version '0.5.19'
 }
 
 apply plugin: 'org.checkerframework'
@@ -94,7 +94,7 @@ the definitions of the custom qualifiers.
 
 ### Specifying a Checker Framework version
 
-Version 0.5.18 of this plugin uses Checker Framework version 3.12.0 by default.
+Version 0.5.19 of this plugin uses Checker Framework version 3.12.0 by default.
 Anytime you upgrade to a newer version of this plugin,
 it might use a different version of the Checker Framework.
 
@@ -196,7 +196,7 @@ plugin to the top-level project. For example:
 
 ```groovy
 plugins {
-  id 'org.checkerframework' version '0.5.18' apply false
+  id 'org.checkerframework' version '0.5.19' apply false
 }
 
 subprojects { subproject ->
@@ -239,7 +239,7 @@ plugins {
   id "net.ltgt.errorprone" version "1.1.1" apply false
   // To do Checker Framework pluggable type-checking (and disable Error Prone), run:
   // ./gradlew compileJava -PuseCheckerFramework=true
-  id 'org.checkerframework' version '0.5.18' apply false
+  id 'org.checkerframework' version '0.5.19' apply false
 }
 
 if (!project.hasProperty("useCheckerFramework")) {

--- a/build.gradle
+++ b/build.gradle
@@ -32,7 +32,7 @@ dependencies {
 }
 
 group 'org.checkerframework'
-version '0.5.18'
+version '0.5.19'
 
 gradlePlugin {
     plugins {

--- a/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
+++ b/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
@@ -292,14 +292,17 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       JavaVersion jvmVersion = JavaVersion.current();
 
       // toolchains are an incubating feature of Gradle as of 6.8.3: https://docs.gradle.org/current/userguide/toolchains.html
-      def toolchain = project.extensions.findByName("java")?.toolchain
-      if (toolchain != null && toolchain.isConfigured()) {
-        def toolchainVersion = toolchain.getLanguageVersion().get()
-        def toolchainVersionInt = Integer.parseInt(toolchainVersion.toString())
-        if (toolchainVersionInt < 8) {
-          throw new IllegalStateException("The Checker Framework does not support Java versions before 8.")
-        } else {
-          jvmVersion = JavaVersion.toVersion(toolchainVersionInt)
+      def javaExtension = project.extensions.findByName("java")
+      if (javaExtension != null && javaExtension.hasProperty("toolchain")) {
+        def toolchain = javaExtension.toolchain
+        if (toolchain != null && toolchain.isConfigured()) {
+          def toolchainVersion = toolchain.getLanguageVersion().get()
+          def toolchainVersionInt = Integer.parseInt(toolchainVersion.toString())
+          if (toolchainVersionInt < 8) {
+            throw new IllegalStateException("The Checker Framework does not support Java versions before 8.")
+          } else {
+            jvmVersion = JavaVersion.toVersion(toolchainVersionInt)
+          }
         }
       }
 

--- a/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
+++ b/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
@@ -291,6 +291,18 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
 
       JavaVersion jvmVersion = JavaVersion.current();
 
+      // toolchains are an incubating feature of Gradle as of 6.8.3: https://docs.gradle.org/current/userguide/toolchains.html
+      def toolchain = project.extensions.findByName("java")?.toolchain
+      if (toolchain != null && toolchain.isConfigured()) {
+        def toolchainVersion = toolchain.getLanguageVersion().get()
+        def toolchainVersionInt = Integer.parseInt(toolchainVersion.toString())
+        if (toolchainVersionInt < 8) {
+          throw new IllegalStateException("The Checker Framework does not support Java versions before 8.")
+        } else {
+          jvmVersion = JavaVersion.toVersion(toolchainVersionInt)
+        }
+      }
+
       // Decide whether to use ErrorProne Javac once configurations have been populated.
       boolean needErrorProneJavac = false
       boolean needJdk8Jar = false

